R/utilities.r

```diff
@@ -104,24 +104,40 @@ remove_missing <- function(df, na.rm = FALSE, vars = names(df), name = "",
   df
 }
 
-finite.cases <- function(x) UseMethod("finite.cases")
 # Returns a logical vector of same length as nrow(x). If all data on a row
 # is finite (not NA, NaN, Inf, or -Inf) return TRUE; otherwise FALSE.
-#' @export
-finite.cases.data.frame <- function(x) {
-  finite_cases <- vapply(x, is.finite, logical(nrow(x)))
+cases <- function(x, fun) {
+  ok <- vapply(x, fun, logical(nrow(x)))
 
   # Need a special case test when x has exactly one row, because rowSums
   # doesn't respect dimensions for 1x1 matrices. vapply returns a vector (not
   # a matrix when the input has one row.
-  if (is.vector(finite_cases)) {
-    all(finite_cases)
+  if (is.vector(ok)) {
+    all(ok)
   } else {
     # Find all the rows where all are TRUE
-    rowSums(as.matrix(finite_cases)) == ncol(x)
+    rowSums(as.matrix(ok)) == ncol(x)
+  }
+}
+
+# Wrapper around is.finite to handle list cols
+is_finite <- function(x) {
+  if (typeof(x) == "list") {
+    !vapply(x, is.null, logical(1))
```

Comment:
Anything could be inside that list (like another list) - being finite doesn't really have a meaning for arbitrary elements so this just drops rows where the list-col is empty.